hasktorch/src/Torch/Tensor.hs

```diff
@@ -221,3 +225,23 @@ instance Show Tensor where
       show2d = showElems show1d (",\n " ++ padding)
       details = "Tensor " ++ (show $ dtype t) ++ " " ++ (show $ shape t) ++ " "
       padding = map (const ' ') details
+
+
+--------------------------------------------------------------------------------
+
+-- Castable instances
+--------------------------------------------------------------------------------
+
+-- NB: ATen only defines Castable [ForeignPtr ATen.Tensor] (ForeignPtr ATen.TensorList)
+instance Castable [Tensor] (ForeignPtr ATen.TensorList) where
+  cast xs f = do
+    ptr_list <- mapM (\x -> (cast x return :: IO (ForeignPtr ATen.Tensor))) xs
+    cast ptr_list f
+  uncast xs f = uncast xs $ \ptr_list -> do
+    tensor_list <- mapM (\(x :: ForeignPtr ATen.Tensor) -> uncast x return) ptr_list
+    f tensor_list
+
+
+instance Castable String (ForeignPtr ATen.StdString) where
+  cast str f = ATen.newStdString_s str >>= f
+  uncast xs f = ATen.string_c_str xs >>= f
```

Comment:
`instance Castable String (ForeignPtr ATen.StdString)` should be in ATen.Cast.
I will move it.